### ISSUE TITLE
Function to compare genotypes stored in `xarray.dataset` using `sgkit`

### ISSUE THREAD

**Issue by szhan** [szhan/tsimpute#94]
This function should help compare genotypes stored in two VCF files, e.g., one from BEAGLE containing imputed genotypes and the other containing ground-truth genotypes. The plan is to develop a version of this function to work for cases where only biallelic or monoallelic sites are compared before generalising it to be incorporated into `sgkit`.

**Comment by hyanwong** [comment]
Does this tie in to https://github.com/tskit-dev/tsinfer/issues/739

On that note, we should probably try to push https://github.com/tskit-dev/tskit/pull/2617 through. I think all the comments are addressed on that PR now, and it can probably be merged, right?

**Comment by szhan** [comment]
Ah, yes! Using `SampleData` to compare genotypes is an absolute nightmare.

**Comment by hyanwong** [comment]
Should be incorporated into SGkit, not SampleData.

**Comment by szhan** [comment]
Yes, I mean using `SampleData` to compare genotypes is inconvenient. Should definitely move onto `sgkit`.

**Comment by szhan** [comment]
If one or both of `ds1` and `ds2` is empty, then a `ValueError` exception should be raised, right? Since there are genotypes to be remapped. h/t @benjeffery for suggesting these tests.

**Comment by szhan** [comment]
Should also add tests involving cases where
1. Some site positions overlap.
2. No site positions are shared.

**Comment by szhan** [comment]
Leaving this open, although it is good enough for now.
